test-collateral/generate-policy/src/main.rs

```diff
@@ -0,0 +1,655 @@
+
+/// About Chihuahua/freestanding-chihuahua/Veracruz.
+const ABOUT: &'static str = "A utility for generating Veracruz JSON policy \
+files from a series of command line arguments.";
+/// The name of the application.
+const APPLICATION_NAME: &'static str = "generate-policy";
+/// The authors list.
+const AUTHORS: &'static str = "The Veracruz Development Team.  See the file \
+`AUTHORS.markdown` in the Veracruz root directory for detailed authorship \
+information.";
+/// The application's version information.
+const VERSION: &'static str = "0.1.0";
+
+/// The single supported ciphersuite embedded in the policy file.
+const POLICY_CIPHERSUITE: &'static str = "TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256";
+
+/// The name of the 'dd' executable to call when computing the hash of the
+/// Mexico City enclave for SGX.
+const DD_EXECUTABLE_NAME: &'static str = "dd";
+/// The name of the 'xxd' executable to call when computing the hash of the
+/// Mexico City enclave for SGX.
+const XXD_EXECUTABLE_NAME: &'static str = "xxd";
+
+/// The default filename of the output JSON policy file, if no alternative is
+/// provided on the command line.
+const DEFAULT_OUTPUT_FILENAME: &'static str = "output.json";
+/// The default expiry of the server certificate (measured in hours, from its
+/// creation) if no alternative is provided on the command line.
+const DEFAULT_CERTIFICATE_EXPIRY: i64 = 8;
+/// The default debug status of the Veracruz enclave, if no alternative is
+/// provided on the command line.
+const DEFAULT_DEBUG_STATUS: bool = false;
+/// The default execution strategy for the WASM binary, if no alternative is
+/// provided on the command line.
+const DEFAULT_EXECUTION_STRATEGY: &'static str = "Interpretation";
+
+////////////////////////////////////////////////////////////////////////////////
+// Command line parsing.
+////////////////////////////////////////////////////////////////////////////////
+
+/// A structure collating all of the arguments passed to the executable.
+#[derive(Clone, Debug, Eq, Hash, Ord, PartialEq, PartialOrd)]
+struct Arguments {
+    /// The filenames of cryptographic certificates associated to each principal
+    /// in the computation.
+    certificates: Vec<String>,
+    /// The roles associated to each principal in the computation.  Note that
+    /// length of the two vectors, `certificates` and `roles`, must match as
+    /// each certificate has an accompanying set of roles to form a compound
+    /// "identity".
+    roles: Vec<Vec<String>>,
+    /// The URL of the Sinaloa instance.
+    sinaloa_url: Option<Url>,
+    /// The URL of the Tabasco instance.
+    tabasco_url: Option<Url>,
+    /// The filename of the Mexico City CSS file for SGX measurement.
+    css_file: String,
+    /// The filename of the Mexico City PRCR0 file for Nitro Enclave
+    /// measurement.
+    pcr0_file: String,
+    /// The filename of the output policy file.
+    output_policy_file: String,
```

Comment:
nit: Should this and other file paths be type `path::PathBuf`?

Reply:
Yes, probably.  Will change.